File: Homeworks/1. Programming/1. C#-Part-1/Other-Solutions/03.Operators-and-Expressions/11. 3rd Bit/thirdBit.cs
/*Problem 11. Bitwise: Extract Bit #3
Using bitwise operators, write a program that uses an expression to find the value of the bit
at index 3 of an unsigned integer read from the console.
The bits are counted from right to left, starting from bit 0.
The result of the expression should be either 1 or 0. Print it on the console.
*/
namespace _11._3rd_Bit
{
    using System;

    class thirdBit
    {
        static void Main()
        {
            int inputNumber = int.Parse(Console.ReadLine());
            int bitPosition = 3;// int.Parse(Console.ReadLine());
            int bitAtPosition = FindNthBit(inputNumber, bitPosition);

            Console.WriteLine(bitAtPosition);
        }

        static int FindNthBit(int number, int bitToFind)
        {
            int bitAtNPosition = (number >> bitToFind) & 1;

            return bitAtNPosition;
        }
    }
}

File: Homeworks/Web Design and UI technlogies/3. JavaScript-Part-1/02.Operators and Expressions/05. Third bit/ThirdBit.js
/**Problem 5. Third bit
 Write a boolean expression for finding if the bit #3 (counting from 0) of a given integer.
 The bits are counted from right to left, starting from bit #0.
 The result of the expression should be either 1 or 0.
 */

function FindThirdBit(number) {
    var position = 3,
        mask = 1 << position,
        nAndMask = number & mask,
        bit = nAndMask >> position;

    return console.log(bit);
}

FindThirdBit(62241);